File: 283-move-zeroes/283-move-zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int temp =0;
        
        for(int i=0,j=0;i<nums.Length&&j<nums.Length;){
            
            
            if(nums[i]==0){
                
                i++;
                
            }
            else{
                temp = nums[j];
                nums[j] = nums[i];
                nums[i] = temp;
                
                j++;
                i++;
                
                
                
                
            }
      
            
            
            
        }
        
    }
}

File: 283-move-zeroes/283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int lastPos = -1;
        
        for (int i = 0; i < nums.size(); i++) {
             if (nums[i] == 0 and lastPos == -1) {
                 lastPos = i;
             }
             else if (nums[i] != 0 and lastPos != -1) {
                 nums[lastPos++] = nums[i];
             }
        }
        
        for (int i = lastPos; i < nums.size(); i++) {
            nums[i] = 0;
        }
    }
};